.github/workflows/ci.yml

```diff
@@ -20,19 +20,35 @@ jobs:
     runs-on: ${{ matrix.os }}
 
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout code
+        uses: actions/checkout@v2
 
       - name: Setup Rust toolchain
-        run: rustup default ${{ matrix.rust_version }}
-
-      - name: Build
-        run: cargo build --all --verbose
-
-      - name: Run tests
-        run: cargo test --all --verbose
-
-      - name: Rustfmt and Clippy
-        run: |
-          cargo fmt -- --check
-          cargo clippy
-        if: matrix.rust_version == 'stable'
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust_version }}
+          override: true
+          components: rustfmt, clippy
+
+      - name: cargo build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+
+      - name: cargo test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+
+      - name: cargo fmt
```

Comment:
Looks like this is failing on rust 1.40.0, maybe we should just run rustfmt on stable in case there are differences between toolchain versions